### DISCUSSION TITLE
How to reorder the columns in a Superset dataset

### DISCUSSION THREAD

**Discussion by Analytics214** [apache/superset#29090] [Q&A / Help]
Stuсked on a problem with reordering columns in tables in new version Superset 4.0.0.
It doesn't work as it worked in previous releases. Earlier we could just drop and change any metrics.
Now we can only delete and place metrics in correct order. It's very complicated and a waste of time.
How can we solve it?
Help please!

**Comment by sfirke** [comment]
Are you talking about a table chart or a dataset? I am having this problem with a dataset. If you mean table chart I'll split off a new issue.

In 4.0.2 I want to order the columns in a dataset, so that when someone does `drill-to-detail by` on a chart, they see the most important columns first. I can't figure this out! I have changed the order of the columns in the data table (it's a SQL view) in the data warehouse, tried sync columns, added and removed a dummy column, etc. and no matter what, the columns appear in what seems like a random order when I do `drill to detail by`.

**Reply by alesnav** [reply]
I'm stucked also in this point... Did you find any solution or workaround?

**Reply by ola-smaha** [reply]
Same issue...

**Reply by sfirke** [reply]
At least the ordering of columns in drill-to-detail may be dealt with during the implementation of SIP-160: https://github.com/apache/superset/issues/32793

**Reply by semenova20081993** [reply]
Thank you for all replies!
The problem was solved with the next version. Now I have 4.0.2 and can easily reorder metrics

**Comment by semenova20081993** [comment]
Uppdate to 4.0.2
